File: mobile/src/main/java/rs/readahead/washington/mobile/domain/repository/IUWAZIServersRepository.java
package rs.readahead.washington.mobile.domain.repository;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;
import rs.readahead.washington.mobile.domain.entity.UWaziUploadServer;

public interface IUWAZIServersRepository {
    Single<List<UWaziUploadServer>> listUwaziServers();
    Single<UWaziUploadServer> createUWAZIServer(UWaziUploadServer server);
    Completable removeUwaziServer(long id);
    Single<UWaziUploadServer> updateUwaziServer(UWaziUploadServer server);
    Single<Long> countUwaziServers();

}

File: mobile/src/main/java/rs/readahead/washington/mobile/domain/repository/uwazi/IUWAZIServersRepository.kt
package rs.readahead.washington.mobile.domain.repository.uwazi

import rs.readahead.washington.mobile.domain.entity.UWaziUploadServer

interface IUWAZIServersRepository {
    suspend fun listUwaziServers(): List<UWaziUploadServer>
    suspend fun createUWAZIServer(server: UWaziUploadServer): UWaziUploadServer?
    suspend fun removeUwaziServer(id: Long)
    suspend fun updateUwaziServer(server: UWaziUploadServer) : UWaziUploadServer?
    suspend fun countUwaziServers(): Long?
}